packages/core-data/src/actions.js

```diff
@@ -543,6 +558,34 @@ export function* saveEntityRecord(
 	}
 }
 
+export function* __experimentalBatchSaveEntityRecords( spec ) {
+	const batch = __experimentalCreateBatch( batchProcessor );
+
+	for ( const { kind, name, recordId, record } of spec ) {
+		if ( recordId ) {
+			dispatch( 'core' ).saveEditedEntityRecord( kind, name, recordId, {
```

Comment:
This uses the global `dispatch`. Probably need to add a control which invokes `dispatch` from the current registry but doesn't wait for it to finish resolving.

```js
yield controls.dispatchAsync( 'core', 'saveEditedEntityRecord', kind, name, recordId, { __experimentalBatch: batch } );
```